pkg/userdata/centos/userdata.go

```diff
@@ -0,0 +1,246 @@
+package_reboot_if_required: true
+{{- end }}
+
+{{ if ne (len .ProviderConfig.SSHPublicKeys) 0 }}
+ssh_authorized_keys:
+{{- range .ProviderConfig.SSHPublicKeys }}
+  - "{{ . }}"
+{{- end }}
+{{- end }}
+
+write_files:
+- path: "/etc/yum.repos.d/kubernetes.repo"
+  content: |
+    [kubernetes]
+    name=Kubernetes
+    baseurl=https://packages.cloud.google.com/yum/repos/kubernetes-el7-$basearch
+    enabled=1
+    gpgcheck=1
+    repo_gpgcheck=1
+    gpgkey=https://packages.cloud.google.com/yum/doc/yum-key.gpg https://packages.cloud.google.com/yum/doc/rpm-package-key.gpg
+
+- path: /etc/sysconfig/selinux
+  content: |
+    # This file controls the state of SELinux on the system.
+    # SELINUX= can take one of these three values:
+    #     enforcing - SELinux security policy is enforced.
+    #     permissive - SELinux prints warnings instead of enforcing.
+    #     disabled - No SELinux policy is loaded.
+    SELINUX=permissive
+    # SELINUXTYPE= can take one of three two values:
+    #     targeted - Targeted processes are protected,
+    #     minimum - Modification of targeted policy. Only selected processes are protected.
+    #     mls - Multi Level Security protection.
+    SELINUXTYPE=targeted
+
+{{- if ne .CloudConfig "" }}
+- path: "/etc/kubernetes/cloud-config"
+  content: |
+{{ .CloudConfig | indent 4 }}
+{{- end }}
+
+- path: "/etc/kubernetes/bootstrap.kubeconfig"
+  content: |
+{{ .Kubeconfig | indent 4 }}
+
+- path: /etc/kubernetes/ca.crt
+  content: |
+{{ .KubernetesCACert | indent 4 }}
+
+- path: "/etc/systemd/system/kubelet.service.d/10-machine-controller.conf"
+  content: |
+    [Unit]
+    Description=Kubelet
+    Requires=network-online.target docker.service
+    After=docker.service network-online.target
+
+    [Service]
+    Restart=always
+    RestartSec=10
+    StartLimitInterval=600
+    StartLimitBurst=50
+    TimeoutStartSec=5min
+    Environment="PATH=/sbin:/bin:/usr/sbin:/usr/bin:/opt/bin"
+    ExecStart=
+    ExecStart=/bin/kubelet \
+      --container-runtime=docker \
+      --cgroup-driver=systemd \
+      --allow-privileged=true \
+      --cni-bin-dir=/opt/cni/bin \
+      --cni-conf-dir=/etc/cni/net.d \
+      --cluster-dns={{ ipSliceToCommaSeparatedString .ClusterDNSIPs }} \
+      --cluster-domain=cluster.local \
+      --network-plugin=cni \
+      {{- if .CloudProvider }}
+      --cloud-provider={{ .CloudProvider }} \
+      --cloud-config=/etc/kubernetes/cloud-config \
+      {{- end }}
+      --cert-dir=/etc/kubernetes/ \
+      --pod-manifest-path=/etc/kubernetes/manifests \
+      --resolv-conf=/etc/resolv.conf \
+      --rotate-certificates=true \
+      --kubeconfig=/etc/kubernetes/kubeconfig \
+      --bootstrap-kubeconfig=/etc/kubernetes/bootstrap.kubeconfig \
+      --lock-file=/var/run/lock/kubelet.lock \
+      --exit-on-lock-contention \
+      --read-only-port 0 \
+      --authorization-mode=Webhook \
+      --anonymous-auth=false \
+      --client-ca-file=/etc/kubernetes/ca.crt
```

Comment:
We should move to `kubeadm join` in the future to get rid of this duplication.
We have the same set of flags in every user-data :/

Reply:
Opened https://github.com/kubermatic/machine-controller/issues/139 for that